platform/m480/src/debug_stdiouart.c

```diff
@@ -0,0 +1,438 @@
+/*
+ * Copyright (c) 2016-2017, Nuvoton, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <uvisor.h>
+
+/* Abstract from M480.h */
```

Comment:
@Patater  According to `uvisor` > `core` > `debug` > `src` > `debug.c`, `default_putc()` is allowed to override. Per our experience, it could help development as uvisor core debug message could emit via USB VCOM rather than semihosting in debug mode. Besides, the platform `beetle` already has the same override.
```
/* During the porting process the implementation of this function can be
 * overwritten to provide a different mechanism for printing messages (e.g. the
 * UART port instead of semihosting). */
#define DEBUG_MAX_BUFFER 128
uint8_t g_buffer[DEBUG_MAX_BUFFER];
int g_buffer_pos;

UVISOR_WEAK void default_putc(uint8_t data)
{
    if (g_buffer_pos < (DEBUG_MAX_BUFFER - 1)) {
        g_buffer[g_buffer_pos++] = data;
```